README.md

```diff
@@ -208,6 +213,183 @@ try {
+} catch (Exception e) {
+	outgoingMessageTracer.error(e.getMessage());
+	Logger.logError(e);
+} finally {
+	outgoingMessageTracer.end();
+}
+```
+
+On the incoming side, we need to differ between the blocking receiving part and processing the received message. Therefore two different tracers are being used: ReceivingMessageTracer and ProcessingMessageTracer.
+
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+// following code is probably inside a busy loop:
+while(true) {
+	ReceivingMessageTracer receivingMessageTracer = oneAgentSDK.traceReceivingMessage(messagingSystemInfo);
+	receivingMessageTracer.start();
+	try {
+		// blocking call - until message is being available:
+		Message queryMessage = theQueue.receive("client queries");
+		ProcessingMessageTracer processingMessageTracer = oneAgentSDK
+			.traceProcessingMessage(messagingSystemInfo);
+		processingMessageTracer.setDynatraceStringTag(
+			queryMessage.getHeaderField(OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME));
+		processingMessageTracer.setVendorMessageId(queryMessage.msgId);
+		processingMessageTracer.setCorrelationId(queryMessage.correlationId);
+		processingMessageTracer.start();
+		try {
+			// do the work ... 
+		} catch (Exception e) {
+			processingMessageTracer.error(e.getMessage());
+			Logger.logError(e);
+		} finally {
+			processingMessageTracer.end();
+		}
+	} catch (Exception e) {
+		receivingMessageTracer.error(e.getMessage());
+		Logger.logError(e);
+	} finally {
+		receivingMessageTracer.end();
+	}
+}
+```
+
+In case of no blocking call happens (e. g. message processing via eventhandler), there is no need to use ReceivingMessageTracer - just trace processing of the message by using the ProcessingMessageTracer:
+
+```Java
+public void onMessage(Message message) {
+	MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+	ProcessingMessageTracer processingMessageTracer = oneAgentSDK
+		.traceProcessingMessage(messagingSystemInfo);
+	processingMessageTracer.setDynatraceStringTag((String)
+		message.getObjectProperty(OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME));
+	processingMessageTracer.setVendorMessageId(queryMessage.msgId);
+	processingMessageTracer.setCorrelationId(queryMessage.correlationId);
+	processingMessageTracer.start();
+	try {
+		// do the work ... 
+	} catch (Exception e) {
+		processingMessageTracer.error(e.getMessage());
+		Logger.logError(e);
+	} finally {
+		processingMessageTracer.end();
+	}
+}
+```
+
+<a name="scav"/>
+
+## Add custom request attributes
+
+You can use the SDK to add custom request attributes to current traced service. Custom request attributes allows you to do easier/better filtering of your requests in dynatrace.
```

Comment:
allows => allow
I miss a description regarding which filtering is supported and for which services. To my knowledge we support this currently only for incoming (web?) services.